qa/os/src/test/java/org/elasticsearch/packaging/util/Docker.java

```diff
@@ -0,0 +1,355 @@
+
+        // The final substring() is because we don't check the directory bit, and we
+        // also don't want any SELinux security context indicator.
+        Set<PosixFilePermission> actualPermissions = fromString(permissions.substring(1, 10));
+
+        assertEquals("Permissions of " + path + " are wrong", actualPermissions, expectedPermissions);
+        assertThat("File owner of " + path + " is wrong", username, equalTo("elasticsearch"));
+        assertThat("File group of " + path + " is wrong", group, equalTo("root"));
+    }
+
+    /**
+     * Waits for up to 5 seconds for a path to exist in the container.
+     */
+    public static void waitForPathToExist(Path path) throws InterruptedException {
+        int attempt = 0;
+
+        do {
+            if (existsInContainer(path)) {
+                return;
+            }
+
+            Thread.sleep(500);
+        } while (attempt++ < 10);
+
+        fail(path + " failed to exist after 5000ms");
+    }
+
+    /**
+     * Perform a variety of checks on an installation. If the current distribution is not OSS, additional checks are carried out.
+     */
+    public static void verifyContainerInstallation(Installation installation, Distribution distribution) {
+        verifyOssInstallation(installation);
+        if (distribution.flavor == Distribution.Flavor.DEFAULT) {
+            verifyDefaultInstallation(installation);
+        }
+    }
+
+    private static void verifyOssInstallation(Installation es) {
+        dockerShell.run("id elasticsearch");
+        dockerShell.run("getent group elasticsearch");
+
+        final Shell.Result passwdResult = dockerShell.run("getent passwd elasticsearch");
+        final String homeDir = passwdResult.stdout.trim().split(":")[5];
+        assertThat(homeDir, equalTo("/usr/share/elasticsearch"));
+
+        Stream.of(
+            es.home,
+            es.data,
+            es.logs,
+            es.config
+        ).forEach(dir -> assertPermissionsAndOwnership(dir, p775));
+
+        Stream.of(
+            es.plugins,
+            es.modules
+        ).forEach(dir -> assertPermissionsAndOwnership(dir, p755));
+
+        // FIXME these files should all have the same permissions
+        Stream.of(
+            "elasticsearch.keystore",
+//            "elasticsearch.yml",
+            "jvm.options"
+//            "log4j2.properties"
+        ).forEach(configFile -> assertPermissionsAndOwnership(es.config(configFile), p660));
+
+        Stream.of(
+            "elasticsearch.yml",
+            "log4j2.properties"
+        ).forEach(configFile -> assertPermissionsAndOwnership(es.config(configFile), p644));
+
+        assertThat(
+            dockerShell.run(es.bin("elasticsearch-keystore") + " list").stdout,
+            containsString("keystore.seed"));
+
+        Stream.of(
+            es.bin,
+            es.lib
+        ).forEach(dir -> assertPermissionsAndOwnership(dir, p755));
+
+        Stream.of(
+            "elasticsearch",
```

Comment:
I don't think we should copy everything that is in Archives. This can (and will) get out of sync. I think we need to think more about how to decouple the checks so they can run agnostically.

Reply:
How about I do that in a follow-up PR? This one has been waiting to merge for a while. I can then take the time to refactor the packaging tests more generally.